--- src/include/thrax/features.h.orig   2018-03-13 17:49:52.514675116 +0000
+++ src/include/thrax/features.h        2018-03-13 17:50:34.955119225 +0000
@@ -459,7 +459,7 @@
         return nullptr;
       }
       int64 label = generated_symbols->Find(featval);
-      if (label == fst::SymbolTable::kNoSymbol) {
+      if (label == fst::kNoSymbol) {
         std::cout << "Feature/value pair " << featval << " is not defined."
                   << std::endl;
         delete generated_symbols;